source/use-cases/local-master-key.txt

```diff
@@ -0,0 +1,89 @@
+
+
+Generate the local **master key** and save it to a file. You should only need to run this once:
+
+.. tabs::
+
+   tabs:
+
+     - id: java-master-key-generator
+       name: "Java"
+       content: |
+
+         .. code-block:: java
+
+           import java.io.FileOutputStream;
+           import java.io.IOException;
+           import java.security.SecureRandom;
+
+           public class CreateMasterKeyFile {
+             public static void main(final String[] args) {
+
+               final byte[] localMasterKey = new byte[96];
+               new SecureRandom().nextBytes(localMasterKey);
+
+               String path = "master-key.txt";
+               try (FileOutputStream stream = new FileOutputStream(path)) {
+                 stream.write(localMasterKey);
+               } catch (IOException e)  {
+                 e.printStackTrace();
+               }
+             }
+           }
+
+     - id: linux-macos-mater-key-generator
+       name: "Linux or MacOS"
+       content: |
+
+         .. code-block:: shell
+
+           echo $(head -c 64 /dev/urandom | base64 | tr -d '\n') >  'master-key.txt'
+
+     - id: windows-10plus-master-key-generator
+       name: "Windows 10+ Powershell"
+       content: |
+
+         .. code-block:: shell
+
+           powershell -command "[Convert]::ToBase64String((1..64|%{[byte](Get-Random -Max 256)}))" | Out-File -FilePath .\master-key.txt
+
+     - id: windows-pre10-master-key-generator
+       name: "Windows 8 and older Powershell"
+       content: |
+
+         .. code-block:: shell
+
+           powershell -command "$r=[byte[]]::new(64);$g=[System.Security.Cryptography.RandomNumberGenerator]::Create();$g.GetBytes($r);[Convert]::ToBase64String($r)" | Out-File -FilePath .\master-key.txt
+
+Next, in our MongoDB client application, add the following logic to read the **master key** from the file we created and construct the KMS settings:
+
+.. code-block:: java
+
+  byte[] masterKey = new byte[96];
+  String path = "master-key.txt";
+  try (FileInputStream fis = new FileInputStream(path)) {
+    masterKey = fis.readAllBytes();
+  } catch (IOException e) {
+    e.printStackTrace();
+  }
+
+  // Specify a local master key in the KMS provider settings
+  Map<String, Map<String, Object>> kmsProviders = new HashMap<String, Map<String, Object>>() {{
+    put("local", new HashMap<String, Object>() {{
+        put("key", localMasterKey);
+    }});
+  }};
+
+The ``kmsProviders`` Map defined above will be used later when specifying the KMS in the encryption settings for the client. At this stage, you should now have a **master key** in a file and code to load it into a Map that will be appended to the configuration of your client.
```

Comment:
"be used" -> Avoid passive voice. You also repeat this information in the last part: "a Map that will be appended to the configuration of your client". Is there a way to merge these?

Consider removing "At this stage" since we are already at the end of the document and the phrase doesn't seem to be conveying much information. "Should" could also go since they just followed the instructions.

Reply:
Thanks, I was attempting to summarize the concepts that they should have picked up by reading through that section (similar to post-conditions). The last sentence talks about both the master key generation as well as setting values in a Map object. Would it be better to:

1) Omit and instead use the summary section at the end of this Use Case document as templated by Nick to : https://github.com/mongodb/docs-ecosystem/commit/4f47fece63aeab1145d7ae5bb8ea6e73ea32d2b6 or otherwise start a new paragraph

2) Create a separate paragraph and reword it to more clearly establish what artifacts the reader has now created

3) Omit the recap info completely

And a slightly off-topic question:
Do we use "we" or "you"? Related discussion here:
https://ux.stackexchange.com/questions/57305/we-vs-you-in-documentation

Reply:
In general you > we.

The rough rule of thumb that I use:
- **we** is for discussing things that the user is provided by the docs
   - e.g. a code block that we walk through - "In this code, we first do x then y."
- **you** is for imperative commands and advisories
   - e.g. a specific direction - "(You) configure the baz to..." (the "you" is often implied in imperative mood)
   - e.g. an optional check - "You should ensure that..."
   - e.g. a note/warning - "You must enable someSetting before foo"